src/pathfinder/yapf/yapf_road.cpp

```diff
@@ -161,8 +161,8 @@ class CYapfCostRoadT
 			int min_speed = 0;
 			int max_veh_speed = v->GetDisplayMaxSpeed();
 			int max_speed = F.GetSpeedLimit(&min_speed);
-			if (max_speed < max_veh_speed) segment_cost += 1 * (max_veh_speed - max_speed);
-			if (min_speed > max_veh_speed) segment_cost += 10 * (min_speed - max_veh_speed);
+			if (max_speed < max_veh_speed) segment_cost += YAPF_TILE_LENGTH * (max_veh_speed - max_speed) * (4 + F.m_tiles_skipped) / max_veh_speed;
```

Comment:
Maybe a little bit late, but where the `4` coming from and why? :)

Reply:
From the rail pathfinder! No, I'm not sure why it's 4 either...